search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityOperationsHandler.kt

```diff
@@ -0,0 +1,60 @@
+package com.egm.stellio.search.web
+
+import com.egm.stellio.search.util.QueryUtils
+import com.egm.stellio.shared.model.*
+import com.egm.stellio.shared.util.*
+import com.egm.stellio.shared.util.JsonLdUtils.addContextsToEntity
+import com.egm.stellio.shared.util.JsonUtils.serializeObject
+import kotlinx.coroutines.reactive.awaitFirst
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.web.bind.annotation.*
+import reactor.core.publisher.Mono
+import java.util.*
+
+@RestController
+@RequestMapping("/ngsi-ld/v1/temporal/entityOperations")
+class TemporalEntityOperationsHandler(
+    private val queryUtils: QueryUtils
+) {
+
+    /**
+     * Partial implementation of 6.24.3.1 - Query Temporal Evolution of Entities With POST
+     */
+    @PostMapping("/query", consumes = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    suspend fun queryEntities(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @RequestBody requestBody: Mono<String>
+    ): ResponseEntity<*> {
+        val body = requestBody.awaitFirst()
+        val params = JsonUtils.deserializeObject(body)
+        val queryParams = LinkedMultiValueMap<String, String>()
+        params.forEach { queryParams.add(it.key, it.value.toString()) }
```

Comment:
since you are recreating the params map from the request body, why not just pass it to `queryUtils.queryTemporalEntities` and do the parsing / extraction in this method or at least why not having a function that does this instead of duplicating the processing in the 2 handlers?

Reply:
I think having two separate functions is cleaner : one that parses and validates the input and one that does the actual search

Reply:
And the function that parses/checks request params should be called by the handler ? then, the handler uses the returned map to call queryTemporalEntities ?

Reply:
seems good.